airmozilla/starred/static/starred/js/star_event.js

```diff
@@ -99,7 +112,10 @@ $(function() {
 
     $('#content').on('click', 'a.star', function () {
         var id = $(this).data('id');
-        $('a.star[data-id=' + id + ']').toggleClass('star-on');
+        $('a.star[data-id=' + id + ']').each(function(i, element) {
+            $(element).toggleClass('star-on');
+            Stars.setToolTip(element);
+        });
```

Comment:
One concern I have is that there's now more stuff in the way of what needs to be fast. The previous line (`$('a.star[data-id=' + id + ']').toggleClass('star-on');`) was good in that it was fast. It doesn't need to do a `.each()` loop thing. Especially since within each loop you have to do that conversion from HTML element to a jQuery element **and** you have to do that `if ($element.hasClass('star-on'))` which is individually fast but when done for 20 elements it adds up. 

Perhaps a much better way would be to re-think the architecture a bit. Perhaps something like:
```javascript
// this is visually important to users on load
$('a.star[data-id=' + id + ']').toggleClass('star-on');  
// these can wait a couple of milliseconds
$('a.star').not('star-on').each(Stars.setTitleOff);  
$('a.star.star-on').each(Stars.setTitleOn);
```